plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -96,36 +101,33 @@ def launch(
 
         num_jobs = len(job_overrides)
         assert num_jobs > 0
+        params = self.params
 
-        # make sure you don't change inplace
-        queue_parameters = self.queue_parameters.copy()
-        OmegaConf.set_struct(queue_parameters, True)
-        executors = {
-            QueueType.auto: submitit.AutoExecutor,
-            QueueType.slurm: submitit.SlurmExecutor,
-            QueueType.local: submitit.LocalExecutor,
-        }
-        init_parameters = {"cluster", "max_num_timeout", "slurm_max_num_timeout"}
-        executor = executors[self.queue](
-            folder=self.folder,
-            **{
-                x: y
-                for x, y in queue_parameters[self.queue.value].items()
-                if x in init_parameters
-            },
-        )
-        executor.update_parameters(
+        # build executor
+        init_params = dict(folder=params.submitit_folder)
+        specific_init_keys = {"max_num_timeout"}
+        init_params.update(
             **{
-                x: y
-                for x, y in queue_parameters[self.queue.value].items()
-                if x not in init_parameters
+                f"{self._EXECUTOR}_{x}": y
+                for x, y in params.items()
+                if x in specific_init_keys
             }
         )
+        init_keys = specific_init_keys | {"submitit_folder"}
+        executor = submitit.AutoExecutor(cluster=self._EXECUTOR, **init_params)
```

Comment:
this line is confusing to me:

The auto executor is expecting the strings "local" or "slurm" as options for cluster?